cmd/fpga_plugin/fpga_plugin.go

```diff
@@ -366,43 +361,25 @@ func main() {
 		debug.Activate()
 	}
 
-	if kubeconfig == "" {
-		config, err = rest.InClusterConfig()
-	} else {
-		config, err = clientcmd.BuildConfigFromFlags(master, kubeconfig)
-	}
+	nodeMode, err := getModeOverrideFromCluster(nodename, kubeconfig, master, mode)
 	if err != nil {
-		fatal(err)
+		fmt.Printf("INFO: could not get mode override from cluster: %+v\n", err)
 	}
```

Comment:
What would happen if getModeOverrideFromCluster fails and nodeMode is empty, i.e. NODE_NAME env is not set?

Reply:
I mean shouldn't we exit here or check if nodeMode is valid?

Reply:
`mode` gets used (it's set to `af` by default).

Reply:
That makes sense to me now, thanks.